ssh/tailssh/incubator.go

```diff
@@ -312,6 +290,143 @@ func beIncubator(args []string) error {
 	return err
 }
 
+// tryLoginShell attempts to handle the ssh session by creating a full login
+// shell. If it was able to do so, it returns true plus any error from running
+// that shell. If it was unable to do so, it returns false, nil.
+//
+// We prefer to create a login shell using either the login command
+// (e.g. /usr/bin/login) or using the su command (e.g. /usr/bin/su). A login
+// shell has the advantage of running PAM authentication, which will set up the
+// connected user's environment.
+//
+// login is preferred over su because it supports the `-h` option, allowing the
+// system to record the remote IP associated with the login.
+//
+// However, login is subject to some limitations.
+//
+// 1. login cannot be used to execute commands except on macOS.
+// 2. On Linux and BSD, login requires a TTY to keep running.
+//
+// Unlike login, su often does not require a TTY, so on Linux hosts that have a
+// su command which accepts the right flags, we fall back to using that when no
+// TTY is available.
+//
+// Note - one nuance of this is that when we use login with the -h option, the
+// shell will use the "remote" PAM profile. When we fall back to using "su",
+// the shell will use the "login" PAM profile.
+func tryLoginShell(logf logger.Logf, ia incubatorArgs) (bool, error) {
+	euid := os.Geteuid()
+	runningAsRoot := euid == 0
+
```

Comment:
Previously, this logic was spread across two processes and 3 different functions. Now, it's in one place.

Also, I pulled a lot of this into the new `tryLoginShell` function to make `beIncubator` shorter.